### PR TITLE
v2: add multi-user manifests

### PR DESCRIPTION
**Pull request by yhwang**
**Description of your changes:**
Add multi-user manifests and remove cache and metadata-writer.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ resources:
 - ../../third-party/tekton-custom-task
 - ../../third-party/minio/base
 - ../../third-party/mysql/base
+- ../../third-party/kfp-csi-s3
```
Comment (Tomcli) [inline manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml:11]: Do we need kfp-csi-s3 for v2? This was mainly meant for copying big artifacts into s3. Do we have any pvc implementation that uses this csi driver by default in v2?

Comment (yhwang) [reply manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml:11]: no built-in usage for that. but you can create/user volume in your pipeline by using kfp-kubernetes sdk

Comment (yhwang) [reply manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml:11]: I guess we can put the manifest in, but not have it in the kustomization.yaml

Comment (Tomcli) [reply manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml:11]: We shouldn't include it by default then, in iks the cos driver has to install separately. If nothing is using it by default we should make users to opt-in if they want to install kfp-csi-s3.

Comment (Tomcli) [reply manifests/kustomize/env/platform-agnostic-tekton/kustomization.yaml:11]: One of the main reason is that cos driver only support iam auth, we can't use default minio auth on cos driver.

**Pull request merged**